versions/unversioned/guides/using-apollo.rst

```diff
@@ -0,0 +1,177 @@
+`react-apollo <https://www.npmjs.com/package/react-apollo>`_ React integration
+package, and the `graphql-tag <https://www.npmjs.com/package/graphql-tag>`_
+library for constructing query documents.
+
+.. code-block:: bash
+
+   npm install apollo-client react-apollo graphql-tag --save
+
+These packages are modular building blocks that can be used independently in
+other environments.
+
+2. Client Setup
+^^^^^^^^^^^^^^^
+
+To start using Apollo we need to create an ``ApolloClient`` and serve that
+client to our React application with an ``ApolloProvider``. The ``ApolloClient``
+is what controls all your GraphQL data and the ``ApolloProvider`` wires the
+client into your React component hierarchy.
+
+Creating a client
+"""""""""""""""""
+
+Initialize an instance of the ``ApolloClient`` imported from the
+`apollo-client <http://npmjs.com/apollo-client>`_ package and provide the client
+with a ``networkInterface`` that points to your GraphQL API.
+
+.. code-block:: javascript
+
+    import ApolloClient, { createNetworkInterface } from 'apollo-client';
+
+    // Replace http://my-api.graphql.com with your GraphQL API’s URL.
+    const client = new ApolloClient({
+      networkInterface: createNetworkInterface({ uri: 'http://my-api.graphql.com' }),
+    });
+
+Creating a provider
+"""""""""""""""""""
+
+Next you will want to connect your client to the rest of your React app with
+``ApolloProvider`` from the
+`react-apollo <https://www.npmjs.com/package/react-apollo>`_ package.
+
+.. code-block:: javascript
+
+    import ApolloClient from 'apollo-client';
+    import { ApolloProvider } from 'react-apollo';
+
+    // Your client from the last step.
+    const client = new ApolloClient({ ... });
+
+    class App extends React.Component {
+      render() {
+        return (
+          <ApolloProvider client={client}>
+            <View>
+              <Text>Hello, Exponent!</Text>
+            </View>
+          </ApolloProvider>
+        );
+      }
+    }
+
+3. Connecting data to a component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All that is left to do is connect data from your GraphQL API to a React
+component where you can render your data to a user. To do this you can use the
+``graphql`` function from the
+`react-apollo <https://www.npmjs.com/package/react-apollo>`_ package to
+create a higher order component.
+
+To learn more about the ``graphql`` and ``gql`` functions used below read the
+Apollo documentation on
+`connecting data <http://dev.apollodata.com/react/higher-order-components.html>`_.
+
+.. code-block:: javascript
+
+    import React from 'react';
+    import gql from 'graphql-tag';
+    import { graphql } from 'react-apollo';
+
+    // The data prop, which is provided by the wrapper below contains,
+    // a `loading` key while the query is in flight and posts when it is ready
+    function PostList({ data: { loading, posts } }) {
+      if (loading) {
+        return <div>Loading</div>;
+      } else {
+        return (
+          <ul>
```

Comment:
This isn't React native code AFAIK. might be easy to steal from Learn Apollo React Native example